drools-ansible-rulebook-integration-api/src/main/java/org/drools/ansible/rulebook/integration/api/domain/constraints/NegatedExistsField.java

```diff
@@ -0,0 +1,126 @@
+package org.drools.ansible.rulebook.integration.api.domain.constraints;
+
+import org.drools.ansible.rulebook.integration.api.domain.RuleGenerationContext;
+import org.drools.ansible.rulebook.integration.api.domain.conditions.ConditionExpression;
+import org.drools.ansible.rulebook.integration.api.rulesmodel.ParsedCondition;
+import org.drools.model.functions.Function1;
+import org.drools.model.prototype.PrototypeExpression;
+import org.drools.model.prototype.PrototypeVariable;
+import org.kie.api.prototype.Prototype;
+import org.kie.api.prototype.PrototypeFactInstance;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+
+import static org.drools.ansible.rulebook.integration.api.domain.conditions.ConditionExpression.map2Expr;
+import static org.drools.model.prototype.PrototypeExpression.thisPrototype;
+
+public enum NegatedExistsField implements RulebookOperator, ConditionFactory {
+
+    INSTANCE;
+
+    public static final String EXPRESSION_NAME = "IsNotDefinedExpression";
+
+    private static final Object ADMITTED_UNDEFINED_VALUE = AdmittedUndefinedValue.INSTANCE;
+
+    @Override
+    public <T, V> BiPredicate<T, V> asPredicate() {
+        return (t, v) -> v == ADMITTED_UNDEFINED_VALUE;
+    }
+
+    @Override
+    public String toString() {
+        return "NEGATED_EXISTS_FIELD";
+    }
+
+    @Override
+    public ParsedCondition createParsedCondition(RuleGenerationContext ruleContext, String expressionName, Map<?, ?> expression) {
+        PrototypeExpression rightExpression = map2Expr(ruleContext, expression).getPrototypeExpression();
+        return new ParsedCondition(thisPrototype(), this, new PrototypeExpressionWithAdmittedUndefined(rightExpression));
+    }
+
+    private static class PrototypeExpressionWithAdmittedUndefined implements PrototypeExpression {
+        private final PrototypeExpression delegate;
+
+        private PrototypeExpressionWithAdmittedUndefined(PrototypeExpression delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public Function1<PrototypeFactInstance, Object> asFunction(Prototype prototype) {
+            return delegate.asFunction(prototype).andThen( result -> result == Prototype.UNDEFINED_VALUE ? ADMITTED_UNDEFINED_VALUE : result );
```

Comment:
This is the main fix and it is necessary to workaround the fact that on the Drools side it prevents any constraint evaluation, returning false, when it meets an `UNDEFINED_VALUE`. I preferred to avoid modifying Drools and fix the problem only here. However I believe that, with this trick, it is not bad that an undefined value is actually evaluated only for this constraint (which is the only one that we know at the moment where this is necessary) and not for each and every constraint that at that point will have to deal with it for no reason.